src/irmin-pack/unix/mapping_file_intf.ml

```diff
@@ -50,6 +50,14 @@ module type S = sig
       Creates temporary files in [root] that are unlinked before the function
       returns. *)
 
+  val create_rev :
+    root:string ->
+    generation:int ->
+    register_entries:(register_entry:(off:int63 -> len:int -> unit) -> unit) ->
+    (t, [> Errs.t ]) result
+  (** Same as [create], but expects [register_entries] to produce live entries
+      in decreasing offset order. *)
```

Comment:
I think it would be nice to not have this kind of enforced order, but if we keep it, I think it would be nice to enforce it internally. Perhaps using the current `current_entry` ref to check that offsets are monotonically decreasing and throwing an exception if the invariant is violated.

Reply:
Thanks, I added an assertion to check for the reverse sorted order! It would indeed be possible to detect that the offsets need to be sorted instead of crashing, but I think if would open the door for an accidental de-optimization in the GC traversal (and the sorting code would remain dead code meanwhile)
What do you think? I'm open to adding the failproof strategy :)

Reply:
Since the scope of usage for this is internal to `irmin-pack`, I think an assertion is a good strategy since it will alert us if we ever break our own code unintentionally. I'd only strongly advocate for a failback sort if the mapping file was an externally facing thing, so I think this is in a great spot now!